src/main/java/teammates/ui/webapi/GetCoursesAction.java

```diff
@@ -38,21 +41,90 @@ public JsonResult execute() {
         String entityType = getNonNullRequestParamValue(Const.ParamsNames.ENTITY_TYPE);
         switch (entityType) {
         case Const.EntityType.STUDENT:
-            return getStudentCourses();
+            if (!isAccountMigrated(userInfo.id)) {
+                return getStudentCourses();
+            }
+
+            return getSqlStudentCourses();
```

Comment:
isAccountMigrated flag does not say anything about the migration status of the course.

Correct flow should be:
1. Get all courses in sql db
2. Get all courses in datastore and filter by isMigrated flag
3. Combine results

Reply:
Hey Samuel, it seems like this way of doing we're just getting all the courses that are migrated. But how about if we are calling this action from the datastore pov?

Reply:
Not sure what you are trying to say, feel free to elaborate if required.

GetCoursesAction should get all courses, migrated or not.